tasks/main.yml

```diff
@@ -9,42 +9,56 @@
     createhome: true
     home: "{{ caddy_home }}"
 
+- name: Build headers to use when making requests to github
+  set_fact:
+    caddy_github_headers: "{{ caddy_github_headers | combine({'Authorisation': 'token ' + caddy_github_token}) }}"
+  when: caddy_github_token | length > 0
+
 - name: Get all Caddy releases
   get_url:
     url: https://api.github.com/repos/mholt/caddy/git/refs/tags
     dest: "{{ caddy_home }}/releases.txt"
     force: true
+    headers: '{{ caddy_github_headers }}'
   retries: 3
   delay: 2
   when: caddy_update
   register: caddy_releases_cache
 
 - name: Set Caddy features
   copy:
-    content: "{{ caddy_features }}"
+    content: "{{ ','.join(caddy_packages) }}"
     dest: "{{ caddy_home }}/features.txt"
   when: caddy_update
   register: caddy_features_cache
 
+- name: Set some common parameters
+  set_fact:
+    caddy_bin_name: "caddy_{{ caddy_os }}_{{ go_arch }}"
```

Comment:
Maybe I overlooked something, but why not just use caddy instead of caddy_{{ caddy_os }}_{{ go_arch }}? I don't think you use it in caddy_url. Then we wouldn't need caddy_bin below.

Reply:
Yes this seems sensible so I have made this change - I don't think the additional complexity is really providing any value here.